qutebrowser/browser/webengine/webenginesettings.py

```diff
@@ -275,23 +288,66 @@ def _update_settings(option):
-    global default_profile, private_profile
-
-    default_profile = QWebEngineProfile.defaultProfile()
-    default_profile.setter = ProfileSetter(default_profile)
-    default_profile.setCachePath(
-        os.path.join(standarddir.cache(), 'webengine'))
-    default_profile.setPersistentStoragePath(
-        os.path.join(standarddir.data(), 'webengine'))
-    default_profile.setter.init_profile()
-    default_profile.setter.set_persistent_cookie_policy()
+def create_profile(storage_name=None, private=False):
+    """Create a QWebEngineProfile and install all required extensions on it
 
-    private_profile = QWebEngineProfile()
-    private_profile.setter = ProfileSetter(private_profile)
-    assert private_profile.isOffTheRecord()
-    private_profile.setter.init_profile()
+    Args:
+        storage_name: Name of the persistent storage for this profile. Must be
+                      unique across profiles. This name is ignored for private
+                      profiles and, if omitted but private=False, the default
+                      profile will be returned.
+        private: If True, an off-the-record profile will be created and
+                 storage_name will be ignored.
+    """
+    if private:
+        profile = QWebEngineProfile()
+        assert profile.isOffTheRecord()
+    elif storage_name:
+        profile = QWebEngineProfile(storage_name)
+        # TODO: set custom paths for cache and storage?
+    else:
+        profile = QWebEngineProfile.defaultProfile()
+        profile.setCachePath(
+            os.path.join(standarddir.cache(), 'webengine'))
+        profile.setPersistentStoragePath(
+            os.path.join(standarddir.data(), 'webengine'))
+    profile.setter = ProfileSetter(profile)
+    profile.setter.init_profile()
+
+    # Install qute scheme handler
+    profile.installUrlSchemeHandler(b'qute', _qute_scheme_handler)
+    if qtutils.version_check('5.11', compiled=False):
+        # WORKAROUND for https://bugreports.qt.io/browse/QTBUG-63378
+        profile.installUrlSchemeHandler(b'chrome-error', _qute_scheme_handler)
+        profile.installUrlSchemeHandler(b'chrome-extension',
+                                        _qute_scheme_handler)
+
+    # Install request interceptor
+    profile.setRequestInterceptor(_req_interceptor)
+
+    # Install download manager
+    profile.downloadRequested.connect(_download_manager.handle_download,
+                                      Qt.DirectConnection)
+
+    # Install cookie filter
+    # On Qt < 5.11, the cookie filter isn't installed.
+    try:
+        profile.cookieStore().setCookieFilter(cookies.accept_cookie)
+    except AttributeError:
+        pass
+
+    # Clear visited links on web history clear
+    hist = objreg.get('web-history')
+    hist.history_cleared.connect(profile.clearAllVisitedLinks)
+    hist.url_cleared.connect(lambda url, profile=profile:
+                             profile.clearVisitedLinks([url]))
```

Comment:
Should I call `global_settings.init_settings()` so settings that could've been applied to older profiles are copied to the new one?

Reply:
Hmm, I'm not sure whether it's a good idea to re-apply all settings on existing profiles whenever a new profile is added - but those should definitely be consistent across profiles.

Maybe you could create a temporary `WebEngineSettings` object which only applies the settings to the newly created profile, and after that, `global_settings will take care on updating it for any future changes?

Reply:
I'm not really familiar with all the "settings" part of the code but I think I get what you mean. I'll take care of it in the next commits.